core/src/main/scala/com.snowplowanalytics.snowplow.collectors.scalastream/CollectorService.scala

```diff
@@ -477,14 +475,25 @@ class CollectorService(
     * Creates an Access-Control-Allow-Origin header which specifically allows the domain which made
     * the request
     * @param request Incoming request
-    * @return Header allowing only the domain which made the request or everything
+    * @param domainRestriction A list of domains which the ACAO header will only be served for, or `None`
+    * if there are no restrictions.
+    * @return A singleton list containing the specified header. If it violates the restrictions, the list
+    * will be empty. If the request had no `Origin` header, the response header will permit every domain.
     */
-  def accessControlAllowOriginHeader(request: HttpRequest): HttpHeader =
-    `Access-Control-Allow-Origin`(request.headers.find {
-      case `Origin`(_) => true
-      case _           => false
+  def accessControlAllowOriginHeader(
+    request: HttpRequest,
+    restrictDomainsTo: Option[List[String]],
+  ): List[HttpHeader] =
```

Comment:
Do we need to change the return type to a `List` here? What is the downside of returning `Access-Control-Allow-Origin(HttpOriginRange.*)` in case all domains should be allowed and `Access-Control-Allow-Origin(HttpOriginRange.Default(domains))` if only certain domains should be allowed?